packages/server/lib/middleware/ratelimit.middleware.ts

```diff
@@ -4,67 +4,95 @@ import { RateLimiterMemory, RateLimiterRedis, RateLimiterRes } from 'rate-limite
 import { createClient } from 'redis';
 
 import { getRedisUrl } from '@nangohq/shared';
-import { flagHasAPIRateLimit, getLogger } from '@nangohq/utils';
+import { flagHasAPIRateLimit, flagHasPlan, getLogger } from '@nangohq/utils';
 
 import type { RequestLocals } from '../utils/express';
+import type { DBPlan } from '@nangohq/types';
 import type { NextFunction, Request, Response } from 'express';
+import type { RateLimiterAbstract } from 'rate-limiter-flexible';
 
 const logger = getLogger('RateLimiter');
 
-const rateLimiter = await (async () => {
+const defaultLimit = parseInt(process.env['DEFAULT_RATE_LIMIT_PER_MIN'] || '0') || 3500;
+const rateLimiterSize: Record<DBPlan['api_rate_limit_size'], number> = {
+    s: defaultLimit / 2,
+    m: defaultLimit,
+    l: defaultLimit * 5,
+    xl: defaultLimit * 10,
+    '2xl': defaultLimit * 50,
+    '3xl': defaultLimit * 100
+};
+const limiter = new Map<DBPlan['api_rate_limit_size'], RateLimiterAbstract>();
```

Comment:
I would use plural `limiters` since it is a map of several limiters

Reply:
indeed